### ISSUE TITLE
[proposal] Frame Presets

### ISSUE THREAD

**Issue by kaerhon** [mypaint/mypaint#288]
I saw Deevad's feature proposal about line pressure presets, i was thinking that it could be nice to have some frame size presets too, to be able to quickly start a project with a specific frame size, without having to write it each time

for exemple, all main screen and paper sizes (like 4/3 ans 16/9 screens, with the old 640x480, 800x600, 1024x768, 1600x1200, etc, the newer 720/1080p and intermediates, and all A6 A5 A4 A3, B6 B5 B4, etc formats)

With, maybe some possibility to add custom presets to the list? as i see it, it could work well with a simple (user-wise) dropdown menu in frames options window, with the list of formats, and maybe some "add custom" at the top of the list, that could save the curent values as a preset. 

what do you think?

**Comment by achadwick** [comment]
I don't have time to even discuss this before the 1.2 release. However I'm fiddling with some very incomplete code that does this via the frame size button: it was mentioned in #217:

![preview screencap of a frame sizing dialog I'm working on](https://cloud.githubusercontent.com/assets/61299/6360053/1f7376a8-bc6f-11e4-8aa2-9376bd9f90ba.png)

That what you want, or are you after something like Krita's way of starting a new project from a template? I don't want to do that because MyPaint is an infinite canvas application.

**Comment by 0ion9** [comment]
That looks good to me for frame preset selection.

I want to point out we don't have to choose between 
"Krita's way of starting a new project from a template"
and "MyPaint is an infinite canvas application". Certainly, the default new document should require no interaction at all. 

But we could also have a menu item 'New from template' which simply loads a document from a list (list is of all documents in a certain directory -- $MYPAINTCONFDIR/templates, perhaps ?) -- and then clears the filename and marks the document dirty. 
Since for many types of work you do have to target specific output dimensions and other picture characteristics, this could be extremely helpful for these workflows.

Thinking further ahead, it could be helpful if we could make the basic 'New' function support full-document templates generally. I have drawn hundreds of icons recently, *not* in MyPaint, and this is one of the main things that would convince me that MyPaint could be practical for that purpose. 
I guess that could be done via preferences -- keeping it simple, a small interface that has a radio button and a file browse button, like this:

New document is created:
* As generic blank document
* From template document : [open]

Anyway I admit I'm getting offtopic here, I can split this into its own issue if you want. I believe it would be a pretty simple and very helpful feature that is also pretty simple to implement, and I'm fully prepared to implement it in its entirety.

**Comment by iirelu** [comment]
Maybe it could go here?
![b](https://cloud.githubusercontent.com/assets/3432543/7080156/7fd78aba-df25-11e4-8b7b-bcb005411cff.png)

**Comment by achadwick** [comment]
@kaerhon Yes, please make a separate issue asking for a New from Template command. Of course, someone will have to make a library of blank .ora templates (or more usefully for source control: unpacked ORA-directories that get built). Could be big, better in a submodule maybe...

@iirelu Yep, that's the plan for where this UI would go. Hang it off the big frame size button in the Edit Frame panel, or an on-canvas button (another UI direction decision that needs to be made)

**Comment by 0ion9** [comment]
@achadwick : Done. I'm not kaerhon though :)

**Comment by odysseywestra** [comment]
How about just add a simple drop down list that that has those presets. That way the frame tool won't be as cluttered, and if they want to add a custom preset just add it to MyPaint's config file. That way we wouldn't have to create a bunch of blank ora files which is unessesary bloat.

**Comment by odysseywestra** [comment]
@kaerhon Could you move this thread over to https://community.mypaint.org/c/feature-requests? I'm working on cleaning up the issue tracker by move all feature requests there so we can get community feedback. If you can also set your post as a wiki that way we can edit the scope of the feature request that would be awesome as well. Don't forget to link back to here as well so I can keep track of it incase we need to reopen the issue. Thanks.
